### DISCUSSION TITLE
Gzip Compression

### DISCUSSION THREAD

**Discussion by bkervaski** [node-fetch/node-fetch#1556] [Q&A] [answered]
In the docs it states the compression is true by default, does this mean that any body sent is automatically compressed with gzip? I ask, because nginx isn't getting a gzip encoding header with 2.6.7, just want to verify before I go bug hunting. Thanks!

**Comment by bkervaski** [comment]
I verified that the request is going out gzipped by dumping the request to the console so I'm missing something obvious, disregard, doesn't look like a node-fetch issue at all.

**Comment by jimmywarting** [comment, marked as answer]
when we say it supports compression then it means that we will send out a default request header `Accept-Encoding: gzip, deflate, br` and the server that accepts the request then have the possibility to respond back with a compressed response. We will then decompress the response and give you the uncompressed response back. 

it dose not mean that we will compress the request body that are being sent to a server

**Comment by bkervaski** [comment]
Oh, I see.  I was dumping the response object, not the body.  So I'll have to gzip the body.

Thanks :)
